src/Cache/FileSystemCacheItemPool.php

```diff
@@ -0,0 +1,212 @@
+            throw new ErrorException("Cache folde couldn't be created");
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getItem(string $key): CacheItemInterface
+    {
+        if (!$this->isValidKey($key)) {
+            throw new InvalidArgumentException("The key '$key' is not valid. The key should follow the pattern |^[a-zA-Z0-9_\.! ]+$|");
+        }
+
+        if (!$this->hasItem($key)) {
+            return new TypedItem($key);
+        }
+
+        // Could this cause any issues? Check for result first maybe?
+        return unserialize(file_get_contents($this->cacheFilePath($key)));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getItems(array $keys = []): iterable
+    {
+        $result = [];
+
+        foreach ($keys as $key) {
+            if ($this->isValidKey($key)){
+                throw new InvalidArgumentException("The key '$key' is not valid. The key should follow the pattern |^[a-zA-Z0-9_\.! ]+$|");
+            }
+
+            $result[$key] = $this->getItem($key);
+        }
+
+        return $result;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function hasItem(string $key): bool
+    {
+        $itemPath = $this->cacheFilePath($key);
+
+        if (!file_exists($itemPath)) {
+            return false;
+        }
+
+        $serializedItem = file_get_contents($this->cacheFilePath($key));
+        return unserialize($serializedItem)->isHit();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function clear(): bool
+    {
+        if (!is_dir($this->cachePath)) {
+            return false;
+        }
+
+        foreach (scandir($this->cachePath) as $fileName) {
+            if ($fileName === '.' || $fileName === '..') {
+                continue;
+            }
+
+            // We are not worrying for folders as the cache shouldn't have 
+            // folders inside. Should we continue deleting files on false?
+            if (!unlink("$this->cachePath/$fileName")) {
+                return false;
+            }
+        }
+
+        if (!rmdir($this->cachePath)) {
+            return false;
+        }
+        
+        if (!mkdir($this->cachePath)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function deleteItem(string $key): bool
+    {
+        if ($this->isValidKey($key)) {
+            throw new InvalidArgumentException("The key '$key' is not valid. The key should follow the pattern |^[a-zA-Z0-9_\.! ]+$|");
+        }
+
+        return unlink($this->cacheFilePath($key));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function deleteItems(array $keys): bool
+    {
+        $result = true;
+
+        foreach ($keys as $key) {
+            if (!$this->isValidKey($key)) {
+                throw new InvalidArgumentException("The key '$key' is not valid. The key should follow the pattern |^[a-zA-Z0-9_\.! ]+$|");
+            }
+
```

Comment:
`deleteItem` method is already doing this check. 

```suggestion
```

Reply:
![image](https://github.com/googleapis/google-auth-library-php/assets/9062626/846ee6fc-15a1-484c-9b0b-e3fdd8cc7499)